collector/collector.go

```diff
@@ -356,12 +365,14 @@ type internalMetrics struct {
 }
 
 type Collector struct {
-	ctx     context.Context
-	target  string
-	auth    *config.Auth
-	module  *config.Module
-	logger  log.Logger
-	metrics internalMetrics
+	ctx         context.Context
+	target      string
+	auth        *config.Auth
+	authName    string
+	modules     map[string]*config.Module
```

Comment:
I just noticed that the order in which modules are acquired changes each time because of the use of a map. For some people this may not be the behavior they thought it would be.

Reply:
Oh, yea, maybe we need to make a wrapper type and make it a simple slice.